### PR TITLE
feat(user API): Add private profile

### PR DESCRIPTION
**Pull request by quevenliu**

### CODE REVIEW

**Comment by togimoto** [comment]
What does the last commit message mean?
> feat(User API): private profile not yet debugged

File: src/backend/controllers/users.js [hunk 1]
```diff
@@ -95,14 +95,14 @@ export function updateProfile(req, res) {
     });
 }
 
-export async function getProfile(req, res) {
+async function getPublicProfile(req, res){
```
Comment (togimoto) [inline src/backend/controllers/users.js:98]: @ZaiKuku told me public profile should include all the used books a user is selling. Did you and him agree that this is not needed?

**Review by togimoto — APPROVED** [review]
_This is a test to experiment with Github features. Please ignore this approval._

**Comment by quevenliu** [comment]
> What does the last commit message mean?
> > feat(User API): private profile not yet debugged

我當時還沒測試
但跑起來之後出乎意料的沒有任何bug

File: src/backend/model/users.js [hunk 2]
```diff
@@ -124,9 +161,89 @@ async function getRating(StudentID) {
...
+            INNER JOIN book AS b ON ub.BookID = b.ISBN
+            LEFT JOIN purchase AS p ON ub.UsedBookID = p.UsedBookID
+            WHERE ub.SellerID = $1 AND p.BuyerID IS NULL`,
+            values: [ userID ],
+        };
+    
+        return db.query(query)
+            .then((result) => {
+                for (let i = 0; i < result.rows.length; i++) {
+                    result.rows[i].AskingPrice = parseFloat(result.rows[i].askingprice);
```
Comment (togimoto) [inline src/backend/model/users.js:235]: Why is `askingprice` a float and not an integer as in the schema?

Comment (quevenliu) [reply src/backend/model/users.js]: Thanks for reminding. I fixed it.

**Review by togimoto — APPROVED** [review]
Good job.

**Pull request merged**